From 5d657fe538a05dc3552e1c9f1454053a2abd3ab4 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Fri, 10 Mar 2023 16:36:41 +0100
Subject: [PATCH 0139/1695] drivers: clk: stm32mp15: force
 clk_op_compute_rate() in unpaged memory

This function can be used during low power sequences. Force it into
unpaged memory.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 053ae9bb38e41cbf8620ae891725366363fed2b6 drivers: clk: stm32mp15: force clk_op_compute_rate() in unpaged memory]
Change-Id: I914d89c0d4c349f02fd8f8cb0b099f396331e508
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
---
 core/drivers/clk/clk-stm32mp15.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/core/drivers/clk/clk-stm32mp15.c b/core/drivers/clk/clk-stm32mp15.c
index b2dae29d6..e73419a54 100644
--- a/core/drivers/clk/clk-stm32mp15.c
+++ b/core/drivers/clk/clk-stm32mp15.c
@@ -1474,6 +1474,7 @@ static unsigned long clk_op_compute_rate(struct clk *clk,
 {
 	return _stm32_clock_get_rate(clk_to_clock_id(clk));
 }
+DECLARE_KEEP_PAGER(clk_op_compute_rate);
 
 static TEE_Result clk_op_enable(struct clk *clk)
 {
-- 
2.47.2

